docs/docs/jsx-in-depth.md

```diff
@@ -0,0 +1,300 @@
+---
+id: jsx-in-depth
+title: JSX In Depth
+permalink: docs/jsx-in-depth.html
+---
+
+Fundamentally, JSX just provides syntactic sugar for the `React.createElement(component, props, children)` function. The JSX code:
+
+```js
+<MyComponent prop1={'one'} prop2={'two'}>stuff</MyComponent>
+```
+
+transpiles into:
+
+```js
+React.createElement(MyComponent, {prop1: 'one', prop2: 'two'}, 'stuff')
+```
+
+You can also use the self-closing form of the tag if there are no children. So:
+
+```js
+<div id={'blorp'} />
+```
+
+transpiles into:
+
+```js
+React.createElement('div', {id: 'blorp'}, null)
+```
+
+If you want to test out how some specific JSX is converted into JavaScript, you can try out [the online Babel transpiler](http://tinyurl.com/j26hzbz).
+
+## HTML Tags and React Components
+
+The first part of a JSX tag determines the type of the JSX element.
+
+Capitalized types indicate that the JSX tag is referring to a React component. These tags get transpiled into a direct reference to the named variable, so if you use the JSX `<Foo />` expression, `Foo` must be in scope.
+
+Since JSX transpiles into calls to `React.createElement`, the `React` library must also always be in scope from your JSX code.
+
+For example, both of the imports are necessary in this code, even though 'React' and 'MyComponent' are not directly referenced from JavaScript:
+
+```js
+import React from 'react';
+import MyComponent from './MyComponent';
+
+export default function() {
+  return <MyComponent />;
+}
+```
+
+You can also refer to a React component using dot-notation from within JSX. This is convenient if you have a single module that exports many React components. For example, if you have a `Settings` module that exports a `Settings.DatePicker` component, you can invoke it directly from JSX with:
+
+```js
+import React from 'react';
+import Settings from './settings';
+
```

Comment:
I ditched the import and cut it down to one example instead of two to de-emphasize it. I'd like to keep this here though because this section of the document is explaining "all the ways to specify the element type".